MediaContent/Model/ContentProcessor.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContent\Model;
+
+use Magento\Framework\Exception\IntegrationException;
+use Magento\MediaContentApi\Api\ExtractAssetFromContentInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Process relation managing between media asset and content: assign or unassign relation if exists.
+ */
+class ContentProcessor
+{
+    /**
+     * @var ExtractAssetFromContentInterface
+     */
+    private $extractAssetFromContent;
+
+    /**
+     * @var AssignAsset
+     */
+    private $assignAsset;
+
+    /**
+     * @var GetAssetsUsedInContent
+     */
+    private $getAssetsUsedInContent;
+
+    /**
+     * @var UnassignAsset
+     */
+    private $unassignAsset;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * ContentProcessor constructor.
+     *
+     * @param ExtractAssetFromContentInterface $extractAssetFromContent
+     * @param AssignAsset $assignAsset
+     * @param GetAssetsUsedInContent $getAssetsUsedInContent
+     * @param UnassignAsset $unassignAsset
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ExtractAssetFromContentInterface $extractAssetFromContent,
+        AssignAsset $assignAsset,
+        GetAssetsUsedInContent $getAssetsUsedInContent,
+        UnassignAsset $unassignAsset,
+        LoggerInterface $logger
+    ) {
+        $this->extractAssetFromContent = $extractAssetFromContent;
+        $this->assignAsset = $assignAsset;
+        $this->getAssetsUsedInContent = $getAssetsUsedInContent;
+        $this->unassignAsset = $unassignAsset;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Create new relation between media asset and content or updated existed.
+     *
+     * @param string $contentEntityId
+     * @param array $contentData
+     * @param string $contentType
+     *
+     * @throws IntegrationException
+     */
+    public function execute(string $contentEntityId, array $contentData, string $contentType): void
+    {
+        try {
+            foreach ($contentData as $contentField => $content) {
+                $relations = $this->getAssetsUsedInContent->execute($contentType, $contentEntityId, $contentField);
+                if (empty($relations)) {
```

Comment:
I don't think the system should be a separate logic branch for empty relations - updateRelation method should cover this case